Add more schemas

Will-Lo opened the pull request:
Adds addBroker, demoteBroker, removeBroker, topicConfiguration, and userTasks endpoints to schema.

Also adds a build script and instructions to generate the documentation

kidkun commented:
@Will-Lo  Thanks for the patch. This patch seems failed `RequestParameterTest`, PTAL.

Will-Lo commented:
@kidkun  I'll fix it up today, thanks for the heads up!

kidkun commented on cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java:
```diff
@@ -39,6 +39,9 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
+
```
nit:redundant empty line.

kidkun commented on cruise-control/src/yaml/endpoints/addBroker.yaml:
```diff
@@ -0,0 +1,184 @@
+AddBrokerEndpoint:
+  post:
+    operationId: addBroker
+    summary: Move partitions to the specified brokers, according to the specified goals
```
Can we ensure all the `summary` and `description` fields end with dot for the new `yaml` files added in this PR.

kidkun commented on cruise-control/src/yaml/endpoints/addBroker.yaml:
```diff
@@ -0,0 +1,184 @@
+          type: set
+          items:
+            type: integer
+          # default is a null set if no values are provided
```
it is not allowed to not set brokerid for this endpoint.

kidkun commented on cruise-control/src/yaml/endpoints/addBroker.yaml:
```diff
@@ -0,0 +1,184 @@
+          minimum: 1
+    responses:
+      '200':
+        description: |-
```
(Also apply to other yaml files added in this PR)can we remove the new line here(to be consistent with existing yaml)

kidkun commented on cruise-control/src/yaml/endpoints/addBroker.yaml:
```diff
@@ -0,0 +1,184 @@
+      # Response for all errors
+      default:
+        description: |-
+          Unexpected error
```
(Also apply to other yaml files added in this PR)`Unexpected error` -> `Error response.`(to be consistent with existing endpoints)

kidkun commented on cruise-control/src/yaml/endpoints/demoteBroker.yaml:
```diff
@@ -0,0 +1,147 @@
+          type: set
+          items:
+            type: int32
+            # default is a null set if no values are provided
```
it is not allowed to not set brokerid for this endpoint.

kidkun commented on cruise-control/src/yaml/endpoints/removeBroker.yaml:
```diff
@@ -0,0 +1,193 @@
+          type: set
+          items:
+            type: integer
+          # default is a null set if no values are provided
```
it is not allowed to not set brokerid for this endpoint.

kidkun commented on cruise-control/src/yaml/endpoints/topicConfiguration.yaml:
```diff
@@ -0,0 +1,177 @@
+        description: Regular expression to specify subject topics
+        schema:
+          type: string # topics regex
+          default: null
```
Can we add an example here?

kidkun commented on cruise-control/src/yaml/endpoints/userTasks.yaml:
```diff
@@ -0,0 +1,65 @@
+        in: query
+        description: Comma separated IP addresses to filter the task results Cruise Control report
+        schema:
+          type: string
```
Can we add an example for this field?

kidkun commented on cruise-control/src/yaml/endpoints/userTasks.yaml:
```diff
@@ -0,0 +1,65 @@
+        in: query
+        description: Comma separated endpoints to filter the task results Cruise Control report
+        schema:
+          type: string
```
Can we add an example for this field?

kidkun commented on cruise-control/src/yaml/endpoints/userTasks.yaml:
```diff
@@ -0,0 +1,65 @@
+        schema:
+          type: number
+          format: int64
+          default: MAX_INT
```
Can we add an `minimum` for this field?

kidkun commented on cruise-control/src/yaml/endpoints/userTasks.yaml:
```diff
@@ -0,0 +1,65 @@
+        in: query
+        description: Comma separated HTTP request types to filter the task results Cruise Control report
+        schema:
+          type: string
```
Can we add an example for this field?

Will-Lo commented:
@kidkun I addressed the changes, I also changed `allow_capacity_estimation` to be consistent across all endpoints as well as there was a small discrepancy between some endpoints.

kidkun commented on cruise-control/src/yaml/endpoints/addBroker.yaml:
```diff
@@ -1,31 +1,31 @@
       - name: brokerid
         in: query
-        description: List of target broker ids
+        description: List of target broker ids,
```
nit: `,` -> `.`

kidkun commented on cruise-control/src/yaml/endpoints/topicConfiguration.yaml:
```diff
@@ -4,26 +4,27 @@ TopicConfigurationEndpoint:
         schema:
           type: string # topics regex
           default: null
+          example: topic_[0-9].*
```
nit: can we use URL encoding for this regex, i.e. `topic_%5B0-9%5D.%2A`
